Crash after x hours running during testing autopeering

## Bug description
Testing the autopeering. After 5 hours running the node crashes
## Rust version
Which version of Rust are you running?
- Rust version:
` stable-x86_64-pc-windows-msvc updated - rustc 1.58.1 (db9d1b20b 2022-01-20) (from rustc 1.58.0 (02072b482 2022-01-11))`
## Which version of Bee are you using?
```
PS D:\github\bee\bee-node> ..\target\release\bee.exe -v
0.3.0-rc2-99682d7
PS D:\github\bee\bee-node>
```

## Hardware specification

What hardware are you using?

- Operating system: W10 Prof
- RAM: 32 GB
- Cores: 48 AMD Threadripper 3960X
- Device: SSD

## Steps To reproduce the bug

1. Enable autopeering
2.  When needed add 1 static peer to get synced
3. After x hours the node crashes. Seen it twice now, once after 12 hours, other after 5 hours.

## Expected behaviour
Node should be finding enough peers to keep synced

## Actual behaviour

Crash after x hours.

## Errors
`tokio-runtime-worker' panicked at 'error sending response signal: [10, 32, 133, 47, 198, 27, 9, 63, 90, 178, 253, 236, 228, 11, 47, 209, 146, 16, 186, 3, 121, 180, 161, 205, 112, 192, 118, 49, 238, 23, 181, 98, 5, 215, 18, 52, 10, 19, 10, 7, 112, 101, 101, 114, 105, 110, 103, 18, 8, 10, 3, 117, 100, 112, 16, 162, 114, 10, 29, 10, 17, 99, 104, 114, 121, 115, 97, 108, 105, 115, 45, 109, 97, 105, 110, 110, 101, 116, 18, 8, 10, 3, 116, 99, 112, 16, 240, 121, 26, 12, 57, 52, 46, 50, 49, 50, 46, 51, 49, 46, 50, 52]', C:\Users\user\.cargo\git\checkouts\bee-781893e298f3299f\10914be\bee-network\bee-autopeering\src\discovery\manager.rs:686:14
`
Last ~700 lines from debug-logging and actual crash in attached logfile.
config.toml also attached as .txt
[bee_crash_autopeering_testing.log](https://github.com/iotaledger/bee/files/7918393/bee_crash_autopeering_testing.log)
[config_anon.toml.txt](https://github.com/iotaledger/bee/files/7918410/config_anon.toml.txt)

Tried to update to latest dev but having a build error:
```cmd
error[E0282]: type annotations needed
   --> C:\Users\user\.cargo\registry\src\github.com-1ecc6299db9ec823\libp2p-noise-0.32.0\src\protocol\x25519.rs:221:45
    |
221 |         curve25519_sk.copy_from_slice(&hash.as_ref()[..32]);
    |                                        -----^^^^^^--
    |                                        |    |
    |                                        |    cannot infer type for type parameter `T` declared on the trait `AsRef`
    |                                        this method call resolves to `&T`
    |
    = note: type must be known at this point

   Compiling libp2p-mplex v0.29.0
   Compiling libp2p-yamux v0.33.0
For more information about this error, try `rustc --explain E0282`.
error: could not compile `libp2p-noise` due to previous error
warning: build failed, waiting for other jobs to finish...
error: build failed
```
```cmd
PS D:\github\bee> git show
commit e836fde6de7b9df4bc0a6e447f67b9cab16c76ce (HEAD -> dev, tag: v0.3.0-rc4, origin/dev)
Author: Thibault Martinez <thibault@iota.org>
Date:   Fri Feb 11 18:19:35 2022 +0100
Date:   Fri Feb 11 18:19:35 2022 +0100
Date:   Fri Feb 11 18:19:35 2022 +0100
Date:   Fri Feb 11 18:19:35 2022 +0100
Date:   Fri Feb 11 18:19:35 2022 +0100
Date:   Fri Feb 11 18:19:35 2022 +0100

    bee-node-0.3.0-rc4 (#1131)

diff --git a/Cargo.lock b/Cargo.lock
index 58ad41ad..691578ec 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -315,7 +315,7 @@ dependencies = [

 [[package]]
 name = "bee-node"
-version = "0.3.0-rc3"
+version = "0.3.0-rc4"
 dependencies = [
  "anymap",
  "async-trait",
diff --git a/bee-node/Cargo.toml b/bee-node/Cargo.toml
index daf2deb4..075d06b2 100644
--- a/bee-node/Cargo.toml
+++ b/bee-node/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "bee-node"
-version = "0.3.0-rc3"
+version = "0.3.0-rc4"
 authors = [ "IOTA Stiftung" ]
 edition = "2021"
 description = "Default node implementations."
```

--- a/bee-network/bee-autopeering/src/discovery/manager.rs
+++ b/bee-network/bee-autopeering/src/discovery/manager.rs
@@ -681,9 +681,9 @@ fn handle_verification_response(verif_res: VerificationResponse, verif_reqval: R
 
     // Send the response notification.
     if let Some(tx) = verif_reqval.response_tx {
-        // Panic: we don't allow channel send errors.
-        tx.send(verif_res.to_protobuf().to_vec())
-            .expect("error sending response signal");
+        // Note: if sending fails, then the response arrived a tiny bit too late for the predefined timeout (race
+        // condition), so we just ignore the error in that case.
+        let _ = tx.send(verif_res.to_protobuf().to_vec());
     }
 }
 
@@ -736,8 +736,9 @@ fn handle_discovery_response(disc_res: DiscoveryResponse, disc_reqval: RequestVa
 
     // Send the response notification.
     if let Some(tx) = disc_reqval.response_tx {
-        // Panic: we don't allow channel send errors.
-        tx.send(ctx.msg_bytes.to_vec()).expect("error sending response signal");
+        // Note: if sending fails, then the response arrived a tiny bit too late for the predefined timeout (race
+        // condition), so we just ignore the error in that case.
+        let _ = tx.send(ctx.msg_bytes.to_vec());
     }
 }
 

--- a/bee-network/bee-autopeering/src/discovery/manager.rs
+++ b/bee-network/bee-autopeering/src/discovery/manager.rs
@@ -683,7 +683,9 @@ fn handle_verification_response(verif_res: VerificationResponse, verif_reqval: R
     if let Some(tx) = verif_reqval.response_tx {
         // Note: if sending fails, then the response arrived a tiny bit too late for the predefined timeout (race
         // condition), so we just ignore the error in that case.
-        let _ = tx.send(verif_res.to_protobuf().to_vec());
+        if tx.send(verif_res.to_protobuf().to_vec()).is_err() {
+            log::debug!("Failed to send response notification. Receiver already dropped.");
+        }
     }
 }
 
@@ -738,7 +740,9 @@ fn handle_discovery_response(disc_res: DiscoveryResponse, disc_reqval: RequestVa
     if let Some(tx) = disc_reqval.response_tx {
         // Note: if sending fails, then the response arrived a tiny bit too late for the predefined timeout (race
         // condition), so we just ignore the error in that case.
-        let _ = tx.send(ctx.msg_bytes.to_vec());
+        if tx.send(ctx.msg_bytes.to_vec()).is_err() {
+            log::debug!("Failed to send response notification. Receiver already dropped.");
+        }
     }
 }
 

--- a/bee-network/bee-autopeering/src/peering/manager.rs
+++ b/bee-network/bee-autopeering/src/peering/manager.rs
@@ -410,7 +410,9 @@ fn handle_peering_response<V: NeighborValidator>(
     if let Some(tx) = peer_reqval.response_tx {
         // Note: if sending fails, then the response arrived a tiny bit too late for the predefined timeout (race
         // condition), so we just ignore the error in that case.
-        let _ = tx.send(ctx.msg_bytes.to_vec());
+        if tx.send(ctx.msg_bytes.to_vec()).is_err() {
+            log::debug!("Failed to send response notification. Receiver already dropped.");
+        }
     }
 }
 

--- a/bee-network/bee-autopeering/src/peering/manager.rs
+++ b/bee-network/bee-autopeering/src/peering/manager.rs
@@ -408,7 +408,9 @@ fn handle_peering_response<V: NeighborValidator>(
 
     // Send the response notification.
     if let Some(tx) = peer_reqval.response_tx {
-        tx.send(ctx.msg_bytes.to_vec()).expect("error sending response signal");
+        // Note: if sending fails, then the response arrived a tiny bit too late for the predefined timeout (race
+        // condition), so we just ignore the error in that case.
+        let _ = tx.send(ctx.msg_bytes.to_vec());
     }
 }
 

